src/linear_algebra/multilinear/basis.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2021 Joseph Myers. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joseph Myers
+-/
+import linear_algebra.basis
+import linear_algebra.multilinear.basic
+
+/-!
+# Multilinear maps in relation to bases.
+
+This file proves lemmas about the action of multilinear maps on basis vectors.
+
+-/
+
+open multilinear_map
+
+variables {R : Type*} {ι : Type*} {n : ℕ} {M : fin n → Type*} {M₂ : Type*} {M₃ : Type*}
+variables [comm_semiring R] [add_comm_monoid M₂] [add_comm_monoid M₃] [∀i, add_comm_monoid (M i)]
+variables [∀i, module R (M i)] [module R M₂] [module R M₃]
+
+/-- Two multilinear maps indexed by `fin n` are equal if they are equal when all arguments are
+basis vectors. -/
+lemma basis.ext_multilinear_fin {f g : multilinear_map R M M₂} {ι₁ : fin n → Type*}
+  (e : Π i, basis (ι₁ i) R (M i)) (h : ∀ v : Π i, ι₁ i, f (λ i, e i (v i)) = g (λ i, e i (v i))) :
```

Comment:
How about some extra brackets to make this a bit easier to read (it took me a few moments to parse this mentally):
```suggestion
  (e : Π i, basis (ι₁ i) R (M i)) (h : ∀ v : (Π i, ι₁ i), f (λ i, e i (v i)) = g (λ i, e i (v i))) :
```

Reply:
I'd maybe go for:
```suggestion
  (e : Π i, basis (ι₁ i) R (M i)) (h : ∀ (v : Π i, ι₁ i), f (λ i, e i (v i)) = g (λ i, e i (v i))) :
```
instead